### PR TITLE
Add unittest to check realm object and realm this types

### PR DESCRIPTION
**Pull request by galpeter**
Make sure that realm object and realm this types are tested.

### CODE REVIEW

File: tests/unit-core/test-api-value-type.c [hunk 1]
```diff
@@ -124,6 +124,33 @@ main (void)
...
+  if (jerry_is_feature_enabled (JERRY_FEATURE_REALM))
+  {
+    jerry_value_t new_realm = jerry_create_realm ();
+    jerry_value_t old_realm = jerry_set_realm (new_realm);
+
+    jerry_type_t new_realm_type = jerry_value_get_type (new_realm);
+    TEST_ASSERT (new_realm_type == JERRY_TYPE_OBJECT);
+
+    jerry_value_t new_realm_this = jerry_realm_get_this (new_realm);
+    jerry_type_t new_realm_this_type = jerry_value_get_type (new_realm_this);
```
Comment (zherczeg) [inline tests/unit-core/test-api-value-type.c:136]: By default the `this` is the same as `realm`.

File: tests/unit-core/test-api-value-type.c [hunk 2]
```diff
@@ -124,6 +124,33 @@ main (void)
...
+
+    jerry_type_t old_realm_type = jerry_value_get_type (old_realm);
+    TEST_ASSERT (old_realm_type == JERRY_TYPE_OBJECT);
+
+    jerry_release_value (new_realm);
+
+    /* Restore the old realm as per docs */
+    jerry_set_realm (old_realm);
+
+    jerry_value_t old_realm_this = jerry_realm_get_this (old_realm);
```
Comment (zherczeg) [inline tests/unit-core/test-api-value-type.c:148]: You should not use old_realm after it is restoered.

**Review by zherczeg — CHANGES_REQUESTED** [review]
I would put this into:
https://github.com/jerryscript-project/jerryscript/blob/master/tests/unit-core/test-realm.c

What is the case that is tested here, but not in the test file?

**Comment by galpeter** [comment]
> What is the case that is tested here, but not in the test file?

There is an API method `jerry_value_get_type` which returns the type of a value. There were no tests where a realm's type is queried. Previously there a problem where an value was returned but the API method was unable to query it's type. To avoid such cases it would be good to always add type test for various (at least create) methods.

And all value type tests are in this file.

**Comment by galpeter** [comment]
In addition I've just realized, the `jerry_object_get_type` for realms should also be tested.

File: tests/unit-core/test-api-value-type.c (hunk 1 above)
Comment (galpeter) [reply tests/unit-core/test-api-value-type.c:136]: Okay and? I've explicitly testing it's type.

File: tests/unit-core/test-api-value-type.c (hunk 2 above)
Comment (galpeter) [reply tests/unit-core/test-api-value-type.c:148]: yeah I've wanted to add this before the set_realm call above. Will fix it.

**Comment by galpeter** [comment]
@zherczeg I've updated the PR

**Review by zherczeg — APPROVED** [review]
LGTM

**Review by rerobika — APPROVED** [review]

**Pull request merged**